regex/standard.go

```diff
@@ -0,0 +1,25 @@
+// +build !oniguruma
+
+package regex
+
+import (
+	"regexp"
+)
+
+func MustCompile(str string) *regexp.Regexp {
+	return regexp.MustCompile(str)
+}
+
+func QuoteMeta(s string) string {
+	return regexp.QuoteMeta(s)
+}
+
+func NewRegexpArray(r ...*regexp.Regexp) []*regexp.Regexp {
```

Comment:
Looking at this code, it might make sense to define an opaque `Regexp` type in this package that will either contain a Go regexp pointer or an oniguruma's regexp pointer. This way external code will be able to define variables of this type.

Type second option is to use Go type aliases to bind either Go or Oni regexp to a type name in this package. I think this might be the most elegant solution.